### PR TITLE
Feature/find distinct

### PR DESCRIPTION
**Pull request by swim**
This PR addresses #5163

### CODE REVIEW

File: src/query-builder/SelectQueryBuilder.ts [hunk 1]
```diff
@@ -1044,9 +1044,15 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                 transactionStartedByUs = true;
             }
 
-            this.expressionMap.queryEntity = true;
+            if (!this.expressionMap.selectDistinct)
+                this.expressionMap.queryEntity = true;
```
Comment (pleerock) [inline src/query-builder/SelectQueryBuilder.ts]: can you please explain these lines? Its not obvious why we need to enable this flag

File: src/query-builder/SelectQueryBuilder.ts [hunk 2]
```diff
@@ -1044,9 +1044,15 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
...
+
             const results = await this.executeEntitiesAndRawResults(queryRunner);
 
+            // if entities is empty seed with raw
+            if (results.entities.length === 0)
+                results.entities = results.raw;
```
Comment (pleerock) [inline src/query-builder/SelectQueryBuilder.ts]: also, why do we need this?

File: src/query-builder/SelectQueryBuilder.ts (hunk 1 above)
Comment (swim) [reply src/query-builder/SelectQueryBuilder.ts]: Previously this flag was always set as enabled in getRawAndEntities. When this flag is enabled the id of the selected table is added to the query. This means the selection is no longer distinct. Does this flag need to be enabled by default?

File: src/query-builder/SelectQueryBuilder.ts (hunk 2 above)
Comment (swim) [reply src/query-builder/SelectQueryBuilder.ts]: Ugh yes I should have raised this as a talking point / temporary amendment. This needs a better solution. The results object entities is empty without the queryEntity flag.

Comment (imnotjames) [reply src/query-builder/SelectQueryBuilder.ts:1054]: Can this only be done only when you've set distinct?

File: src/query-builder/SelectQueryBuilder.ts (hunk 1 above)
Comment (imnotjames) [reply src/query-builder/SelectQueryBuilder.ts:1048]: If the Id could be omitted with select would it remove the need for these lines?

**Review by imnotjames — COMMENTED** [review]
Can you rebase so the tests run?

**Comment by imnotjames** [comment]
This has been stale for over 6 months, I'm going to be closing it out but the issue will remain.

**Pull request closed without merging**